presto-main/src/main/java/com/facebook/presto/operator/OutputFactory.java

```diff
@@ -17,11 +17,19 @@
 import com.facebook.presto.spi.Page;
 import com.facebook.presto.spi.plan.PlanNodeId;
 import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.OutputPartitioning;
 
 import java.util.List;
+import java.util.Optional;
 import java.util.function.Function;
 
 public interface OutputFactory
 {
-    OperatorFactory createOutputOperator(int operatorId, PlanNodeId planNodeId, List<Type> types, Function<Page, Page> pagePreprocessor, PagesSerdeFactory serdeFactory);
+    OperatorFactory createOutputOperator(
+            int operatorId,
+            PlanNodeId planNodeId,
+            List<Type> types,
+            Function<Page, Page> pagePreprocessor,
+            Optional<OutputPartitioning> outputPartitioning,
```

Comment:
So this `OutputPartitioning` is only required by `PartitionedOutputOperatorFactory` right? (and later `PrestoSparkOutputFactory`)? Maybe check argument in other `OutputFactory` that this shouldn't present

Reply:
Added a check in the `TaskOutputFactory`. Didn't add for others, as the others are only used in tests.

Reply:
Might still be useful to explicitly add the tests (it's more for convenient for readers of the code :)  )